epimodel/tasks.py

```diff
@@ -0,0 +1,502 @@
+
+from epimodel import Level, RegionDataset, algorithms, imports, read_csv_smart, utils
+from epimodel.exports.epidemics_org import process_export, upload_export
+from epimodel.gleam import Batch, GleamDefinition
+from epimodel.gleam import batch as batch_module
+
+logger = logging.getLogger(__name__)
+
+
+class Configuration(luigi.Config):
+    """
+    That these cannot be overriden from CLI due to how luigi evaluates class
+    parameters.
+
+    These can be changed in the config though and will take effect.
+    """
+
+    input_directory: str = luigi.Parameter(
+        description="Directory with manually created input files"
+    )
+    output_directory: str = luigi.Parameter(
+        description="Directory where tasks are going to produce outputs"
+    )
+    gleam_resample: str = luigi.Parameter(
+        default="1D",
+        description="Subsampling of imported data (see Pandas `df.resample`)",
+    )
+
+
+CONFIG = Configuration()
+
+
+def default_from_config(task_name: str, param_name: str) -> dict:
+    return dict(section=task_name, name=param_name)
+
+
+def _prefix_cfg(filename: str, par_attrib="input_directory") -> str:
+    return (Path(getattr(CONFIG, par_attrib)) / filename).as_posix()
+
+
+class RegionsFile(luigi.ExternalTask):
+    regions = luigi.Parameter(default=_prefix_cfg("regions.csv"))
+
+    def output(self):
+        return luigi.LocalTarget(self.regions)
+
+
+class GleamRegions(luigi.ExternalTask):
+    gleams = luigi.Parameter(default=_prefix_cfg("regions-gleam.csv"))
+
+    def output(self):
+        return luigi.LocalTarget(self.gleams)
+
+
+class RegionsAggregates(luigi.ExternalTask):
+    aggregates = luigi.Parameter(default=_prefix_cfg("regions-agg.yaml"))
+
+    def output(self):
+        return luigi.LocalTarget(self.aggregates)
+
+
+@inherits(RegionsFile, GleamRegions, RegionsAggregates)
+class RegionsDatasetTask(luigi.Task):
+    regions_dataset: str = luigi.Parameter(
+        default=_prefix_cfg("region_dataset.pk", "output_directory")
+    )
+
+    def run(self):
+        regions = self.input()["region_file"].path
+        gleams = self.input()["gleam_regions"].path
+        aggregates = self.input()["aggregates"].path
+        rds = RegionDataset.load(regions, gleams, aggregates)
+        algorithms.estimate_missing_populations(rds)
+        with open(self.regions_dataset, "wb") as ofile:
+            dill.dump(rds, ofile)
+
+    def output(self):
+        return luigi.LocalTarget(self.regions_dataset)
+
+    def requires(self):
+        return {
+            "region_file": self.clone(RegionsFile),
+            "gleam_regions": self.clone(GleamRegions),
+            "aggregates": self.clone(RegionsAggregates),
+        }
+
+    @staticmethod
+    def load_dilled_rds(path: str):
+        with open(path, "rb") as ifile:
+            return dill.load(ifile)
+
+
+class JohnsHopkins(luigi.Task):
+    _output_directory: str = luigi.Parameter(default=CONFIG.output_directory)
```

Comment:
Can we parameterize this by date so that it reruns every day? 
i.e. put an `luigi.DateParameter(default=datetime.date.today())`